mlflow/types/agent.py

```diff
@@ -55,7 +50,7 @@ class ChatAgentMessage(BaseModel):
 
     if IS_PYDANTIC_V2_OR_NEWER:
```

Comment:
can this logic be translated into a single def check_content_and_tool_calls

and move the logic to either inside the function or into pydantic_utils files?

Reply:
Refactored to consolidate the function, but I lean towards keeping the logic here instead of moving it to pydantic_utils, if that's ok.